Assets/HoloToolkit-Examples/Utilities/Scripts/SwapVolume.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule.Tests
+{
+    public class SwapVolume : MonoBehaviour, IInputClickHandler
+    {
+        [SerializeField]
+        private GameObject hideThis;
+        public GameObject HideThis
+        {
+            get
+            {
+                return hideThis;
+            }
+
+            set
+            {
+                hideThis = value;
+            }
+        }
+
+        [SerializeField]
+        private GameObject spawnThis;
+        public GameObject SpawnThis
+        {
+            get
+            {
+                return spawnThis;
+            }
+
+            set
+            {
+                spawnThis = value;
+            }
+        }
+
+        [SerializeField]
+        private bool updateSolverTargetToClickSource = true;
+
+        private SolverHandler solverHandler;
+        private Vector3 defaultPosition;
+        private Quaternion defaultRotation;
+        private bool isOn = false;
+        private GameObject spawnedObject;
+
+        private void Start()
+        {
+            defaultPosition = HideThis.transform.position;
+            defaultRotation = HideThis.transform.rotation;
+        }
+
+        public void OnInputClicked(InputClickedEventData eventData)
+        {
+            if (isOn)
+            {
+                if (spawnedObject != null)
+                {
+                    Destroy(spawnedObject);
```

Comment:
Do we want to Instantiate and destroy the object every time?

Why not just Instantiate and then set active or disable?

Destroy when the monobeahviour get destroyed or disabled